src/main/scala/example/TaskMain.scala

```diff
@@ -0,0 +1,140 @@
+package example
+
+import example.Task1RepartitioningApproach.transformationTask1WithRepartitioningApproach
+import example.Task1SimpleApproach.transformationTask1WithSimpleApproach
+import example.Task1SimpleApproachWithComplications.transformationTask1WithSimpleApproachWithComplications
+import example.Task2RepartitioningApproach.transformationTask2WithRepartitioningApproach
+import example.Task2SimpleApproach.transformationTask2WithSimpleApproach
+import example.Task2SimpleApproachWithComplications.transformationTask2WithSimpleApproachWithComplications
+import org.apache.spark.sql.SparkSession
+
+object TaskMain {
+  def main(args: Array[String]): Unit = {
+    // CREATE SPARK SESSION
+
+    val spark = TaskInitialization.createLocalSparkSession
```

Comment:
Why most of the code here is commented out?

Reply:
Well. The definition of the task is mostly to create unit tests. So main is a placeholder to uncomment one of the datasets (statically or dynamically generated) and one of the methods if you want to experiment with them. Probably it should be commented better.

Reply:
yeah, but for me, if we do UTs we should create data there, as on production the data exists, U just use it.
So, natural split is like that that main has impl. and usage of the data in cluster or UTs

Reply:
I totally agree but:
1. It is still an exercise. 
2. I do not know your production setup. So I do not know how you load your data. And where you store the data. From CSV / Parquet files from S3, from data tables from Spark, from data tables from SQL / NoSQL?

Reply:
agree UTs per methods yes, but we use for example e2eUTs: end to end unit tests, without UTs per methods.
So, we can impl. more complicated data sets t check all in one

Reply:
> I totally agree but:
> 
> 1. It is still an exercise.
> 2. I do not know your production setup. So I do not know how you load your data. And where you store the data. From CSV / Parquet files from S3, from data tables from Spark, from data tables from SQL / NoSQL?

Agree again. We have all the data in s3 but then on top of s3 we have tables, so we do not reach to s3 data but to tables. As I've said on the interview, the offline test tests different thinks, how you improve, that's why the base definition is simple, as I like to se how will you improve based on the propositions. OK?